### DISCUSSION TITLE
GraphQL: `PullRequestChangedFile` missing `previous_filename`

### DISCUSSION THREAD

**Discussion by vpanta** [community/community#34714] [Apps, API and Webhooks]
## Request: 
- Add a way to know which source file was renamed via the `PullRequestChangedFile` data on `PullRequest` in the GraphQL API

## Context:
While the REST API has a [Get Commit endpoint](https://docs.github.com/en/rest/commits/commits#get-a-commit), which returns the set of changes making up the commit, the closest similar GraphQL query I can find is the [`PullRequest`'s `files: PullRequestChangedFileConnection` field](https://docs.github.com/en/graphql/reference/objects#pullrequest).

In comparing these, the REST API's `files` property contains a `previous_filename` for when the file change type is `RENAMED`. However, the GraphQL API's `PullRequestChangedFile` has no similar property, and as such, while we can determine a file rename occurred, we cannot tell what the old name of the file was.

There should be some way to know the previous filename via this API, otherwise we are dependent on the REST API still.

**Comment by reeftcanva** [comment]
Additionally, the REST API has commit and file count limits. If you're building tooling that have to be able to handle large changes, there doesn't seem to be an option covered. You may revert to fetching the entire diff and processing that instead.

Options:
- Using the REST API [list pull request files approach](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests-files)
  >Note: Responses include a maximum of 3000 files. The paginated response returns 30 files per page by default.
- Using the REST API to [get each commit](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#get-a-commit)
  >Note: If there are more than 300 files in the commit diff, the response will include pagination link headers for the remaining files, up to a limit of 3000 files. Each page contains the static commit information, and the only changes are to the file listing.
- When using the REST API [comparing commits approach](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#compare-two-commits)
  - Through testing, with pagination, we've found that there is an **undocumented limit** of 300 for the files list when comparing commits. The files list is only filled on the first page of results.
- **Using `graphql` would be ideal if** `PullRequestChangedFile` for `RENAMED` files had a way to get the `previous_filename`
